server/src/test/java/org/elasticsearch/http/DefaultRestChannelTests.java

```diff
@@ -442,4 +510,41 @@ public RestStatus status() {
             return RestStatus.OK;
         }
     }
+
+    private class TestBytesReference extends BytesReference implements Releasable {
```

Comment:
I think you could shorten this by just using `org.elasticsearch.common.bytes.ReleasablePagedBytesReference` with an empty bytes array and passing your close logic as the `Releasable` in its constructor?

Reply:
I pushed 95efb064e18e85223160854dfcb70d068b277499